### PR TITLE
Feature ci jest 02

### PR DESCRIPTION
**Pull request by theGaryLarson**
Added ts compiler for code checks.
Added jest unit testing and
code coverage (need /docs/coverage folder created in root of repository
Added dependabot.yml file (will configure as we get more info on what we want to track)

Enabled Dependabot in repo settings. (Dependencies, security updates, code scanning & secret scanning)

Will enable Dependabot version updates once dependabot.yml file is merged.
Also, we might want to enable push protection blocking commits that contain secrets

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+
+  CI:
+    uses: ./.github/workflows/ci.yml # reusable workflow file path
+    if: github.event.pull_request.merged == true || github.event_name == 'push'
+
+  CD:
+    uses: ./.github/workflows/cd.yml # reusable workflow file path
+    needs: CI
```
Comment (theGaryLarson) [inline .github/workflows/main.yml:24]: added needs: CI to enesure CI finishes before running CD

File: jest.config.js
Comment (theGaryLarson) [inline jest.config.js:1]: I set up this jest config file to test for typescript. Also, working on getting it to create an html on the repo that we can use with github pages so we can see test successes/failures and code coverage

File: package.json [hunk 2]
```diff
@@ -21,5 +24,10 @@
     "react-dom": "18.2.0",
     "tailwindcss": "3.3.2",
     "typescript": "5.1.3"
+  },
+  "devDependencies": {
+    "@types/jest": "29.5.2",
+    "jest": "29.5.0",
+    "ts-jest": "29.1.0"
```
Comment (theGaryLarson) [inline package.json:31]: Had to add these devDependencies to to typescript code checking

File: tests/mocks/math.ts
Comment (theGaryLarson) [inline tests/mocks/math.ts:1]: Contains a simple add function so I could confirm unit testing is behaving as expected

File: tests/unit/math.test.ts
Comment (theGaryLarson) [inline tests/unit/math.test.ts:1]: A simple unit test for add function in jest to confirm expected behavior

**Review by taylorpapke — APPROVED** [review]
Looks great! Very easy to read and understand. I really like the way the code for jobs is set up in the ci.yml.

**Review by brandondombrowsky — DISMISSED** [review]
LGTM

**Review by taylorpapke — DISMISSED** [review]
Looks good to me

**Comment by brinkbrink** [comment]
I'm fine with a bypass merge here.

**Comment by brinkbrink** [comment]
Resolves #7

**Pull request merged**